### PR TITLE
Abandoned Mech Bay 2: Moderately Better Edition

### PR DESCRIPTION
**Pull request by Erikafox**
## About The Pull Request
Adds a new ruin, the Abandoned Mech Bay. The abandoned mech bay is a semi-spacious installation formerly made to hold 4 mechs, and their crew. The base was designed to resist attack, but in its age is much more approachable.

The air processing cave has a small amount of landmines and turrets, along with the quickest way in. Anyone with a weapon and basic medical supplies could destroy the turrets and enter through air processing.
It is also possible to breach through where the walls are thin, such as at the south and west sides of the base.
Once inside you will have to deal with an assortment of syndicate simplemobs, and another set of turrets. If you can clear the main hallway, it's likely you can get inside the mech bays.
The base has fully mapped atmospherics and power supplies. If you really wanted to go so far, you could clean and live in it.
Loot includes:
2 autolathes
2 Exosuit Fabricators
1 Intact Gygax
1 heavy mech laser
1 mech lmg
2 mech lmg ammo boxes
3 oxygen cans
3 nitrogen cans
Assorted Materials
Anything you can pick up off corpses
A medical aid kit
2 first aid kits
1 toxin aid kit
McKinnon the fox
2 Exosuit Diamond Drills
T2 mech parts
Industrial Welders
6 welding gas masks
Toolbelts
3 Pilot helmets
3 mech pilot suits (the funny NGE ones not spacesuits)
An R&D console
Anything you can get by disassembling the base.

- [x] I affirm that I have tested all of my proposed changes and that any issues found during tested have been addressed.

## Why It's Good For The Game
It's the same mech bay as #1495 but better

Also it gets to be tested this time around

## Changelog

:cl:
add: The Abandoned Mech Bay, a new ruin will start spawning on waste worlds. Beware its defenses, and those who've already found their way in.
/:cl:

### CODE REVIEW

File: code/game/area/areas/ruins/wasteplanet.dm [hunk 1]
```diff
@@ -1 +1,43 @@
 //area defs go here :)
+
+//Abandoned Mechbay Ruin
+
```
Comment (Latentish) [inline code/game/area/areas/ruins/wasteplanet.dm:4]: 10 areas is a little excessive.

File: code/game/area/areas/ruins/rockplanet.dm [hunk 2]
```diff
@@ -5,4 +5,3 @@
 
 /area/mine/rockplanet_nanotrasen
 	name = "Abandoned Mining Facility"
```
Comment (Latentish) [inline code/game/area/areas/ruins/rockplanet.dm:7]: I'd recommend reverting the changes to this file.

File: _maps/map_catalogue.txt [hunk 3]
```diff
@@ -573,3 +572,7 @@ Find the key for using this catalogue in "map_catalogue_key.txt"
 		File name ="_maps\RandomRuins\wasteruins\wasteplanet_unhonorable.dmm"
 		Size = (x = 11)(y = 17)(z = 1)
 		Tags = "No Combat", "Medium Loot", "Shelter", "Hazardous"
+
+		File name = "_maps\RandomRuins\wasteruins\wasteplanet_abandoned_mechbay
+		Size = (x = 45)(y = 47)(z = 1)
+		Tags = "Boss Combat Challenge" "Medium Loot" "Shelter" "Hazardous"
```
Comment (Latentish) [inline _maps/map_catalogue.txt]: commas here

**Review by Latentish — DISMISSED** [review]
image
The gygax is a very stompy mech, and would aid quite a lot to running the ruin. I'd prefer you did spread its parts around the base to add to the salvage loot. That way they're incentivized to not just grab the good shit and go.

Also just a request, you should totally put a floor safe with "secret" documents in them. You could write a funny story or something, make a holotape, idk. It'd add some extra flavour.

Otherwise this is pretty nice.

File: code/game/area/areas/ruins/wasteplanet.dm (hunk 1 above)
Comment (Erikafox) [reply code/game/area/areas/ruins/wasteplanet.dm:4]: somewhere along the line my attempts at cutting down got reverted. Whoops.

**Review by ZephyrTFA — APPROVED** [review]

File: _maps/map_catalogue.txt (hunk 3 above)
Comment (Latentish) [reply _maps/map_catalogue.txt]: This still needs commas. Between the tags.

Comment (Erikafox) [reply _maps/map_catalogue.txt]: image

Comment (Latentish) [reply _maps/map_catalogue.txt]: I saw that and resolved it again, apologies.

**Comment by Latentish** [comment]
Let's try to tm this.

**Comment by spockye** [comment]
image

rouge turf

**Review by Apogee-dev — APPROVED** [review]
Looks good and testing hasn't revealed any remaining issues.

**Pull request merged**